### PR TITLE
Support security groups on AWS ELBs. #306

### PR DESCRIPTION
**Pull request by mgulimonov**
Support security groups on AWS ELBs. #306

### CODE REVIEW

**Comment by ckelner** [comment]
https://github.com/TheWeatherCompany/grid-api/issues/306

**Comment by mgulimonov** [comment]
Completed. AWS SDK was eliminated.

File: src/main/java/org/dasein/cloud/network/AbstractLoadBalancerSupport.java [hunk 1]
```diff
@@ -373,4 +373,9 @@ public void removeLoadBalancerHealthCheck(@Nonnull String providerLoadBalancerId
     public boolean healthCheckRequiresLoadBalancer() throws CloudException, InternalException{
         return true;
     }
+
+    @Override
+    public void attachFirewalls(@Nonnull String providerLoadBalancerId, @Nonnull String... firewallIds) throws CloudException, InternalException {
```
Comment (ckelner) [inline src/main/java/org/dasein/cloud/network/AbstractLoadBalancerSupport.java]: Let's update this to "setFirewalls" per @clstokes comment here: https://github.com/TheWeatherCompany/grid-api/pull/712#discussion_r12293310

File: src/main/java/org/dasein/cloud/network/AbstractLoadBalancerSupport.java [hunk 2]
```diff
@@ -373,4 +373,9 @@ public void removeLoadBalancerHealthCheck(@Nonnull String providerLoadBalancerId
...
+        throw new OperationNotSupportedException("Attaching firewalls have not been implemented for " + getProvider().getCloudName());
```
Comment (ckelner) [inline src/main/java/org/dasein/cloud/network/AbstractLoadBalancerSupport.java]: Let's change "Attaching" to "Setting" here also.

File: src/main/java/org/dasein/cloud/network/LoadBalancerSupport.java [hunk 3]
```diff
@@ -56,6 +49,7 @@
     static public final ServiceAction REMOVE_VMS           = new ServiceAction("LB:REMOVE_VM");
     static public final ServiceAction REMOVE_LOAD_BALANCER = new ServiceAction("LB:REMOVE_LOAD_BALANCER");
     static public final ServiceAction CONFIGURE_HEALTH_CHECK = new ServiceAction("LB:CONFIGURE_HEALTH_CHECK");
+    static public final ServiceAction ATTACH_FIREWALLS     = new ServiceAction("LB:ATTACH_FIREWALLS");
```
Comment (ckelner) [inline src/main/java/org/dasein/cloud/network/LoadBalancerSupport.java]: Let's update this to "SET_FIREWALLS" also

File: src/main/java/org/dasein/cloud/network/LoadBalancerSupport.java [hunk 4]
```diff
@@ -472,4 +467,13 @@
...
     public boolean requiresServerOnCreate() throws CloudException, InternalException;
+
+    /**
+     * Attaches an existing Load Balancer to an existing firewalls
+     * @param providerLoadBalancerId the load balancer ID
+     * @param firewallIds the firewalls
+     * @throws CloudException
+     * @throws InternalException
+     */
+    public void attachFirewalls(@Nonnull String providerLoadBalancerId, @Nonnull String... firewallIds) throws CloudException, InternalException;
```
Comment (ckelner) [inline src/main/java/org/dasein/cloud/network/LoadBalancerSupport.java]: Rename to "setFirewalls" also

File: src/main/java/org/dasein/cloud/network/LoadBalancer.java [hunk 5]
```diff
@@ -93,6 +90,7 @@ static public LoadBalancer getInstance(@Nonnull String ownerId, @Nonnull String
     private int[]                   publicPorts;
     private IPVersion[]             supportedTraffic;
     private Map<String,String>      tags;
+    private String[]                firewallIds;
```
Comment (clstokes) [inline src/main/java/org/dasein/cloud/network/LoadBalancer.java]: Rename this to `providerFirewallIds` to be consistent with other core objects.

**Pull request merged**